aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,400 @@
+          .build());
+      LOG.info("Successfully dropped table {} from Glue", identifier);
+      if (purge && lastMetadata != null) {
+        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        LOG.info("Glue table {} data purged", identifier);
+      }
+      LOG.info("Dropped table: {}", identifier);
+      return true;
+    } catch (EntityNotFoundException e) {
+      LOG.error("Cannot drop table {} because table not found or not accessible", identifier, e);
+      return false;
+    } catch (Exception e) {
+      LOG.error("Cannot complete drop table operation for {} due to unexpected exception", identifier, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Rename table in Glue is a drop table and create table.
+   * @param from identifier of the table to rename
+   * @param to new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    // check new namespace exists
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Cannot rename %s to %s because namespace %s does not exist",
+          from, to, to.namespace());
+    }
+    // keep metadata
+    Table fromTable = null;
+    String fromTableDbName = IcebergToGlueConverter.getDatabaseName(from);
+    String fromTableName = IcebergToGlueConverter.getTableName(from);
+    String toTableDbName = IcebergToGlueConverter.getDatabaseName(to);
+    String toTableName = IcebergToGlueConverter.getTableName(to);
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(fromTableDbName)
+          .name(fromTableName)
+          .build());
+      fromTable = response.table();
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchTableException(e, "Cannot rename %s because the table does not exist in Glue", from);
+    }
+
+    dropTable(from, false);
+    // use the same Glue info to create the new table, pointing to the old metadata
+    TableInput.Builder tableInputBuilder = TableInput.builder()
+        .owner(fromTable.owner())
+        .tableType(fromTable.tableType())
+        .parameters(fromTable.parameters());
+    try {
+      glue.createTable(CreateTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(toTableDbName)
+          .tableInput(tableInputBuilder.name(toTableName).build())
+          .build());
+      LOG.info("Renamed table from {} to {}", from, to);
+    } catch (GlueException e) {
+      // rollback, recreate old table and rethrow exception
+      LOG.error("Fail to rename table from {} to {}, rollback the old table", from, to, e);
+      glue.createTable(CreateTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(fromTableDbName)
+          .tableInput(tableInputBuilder.name(fromTableName).build())
+          .build());
+      throw e;
+    }
```

Comment:
Seems like it would be safer to do this in the opposite order: create the new table as a copy and then drop the existing table. That way, if anything goes wrong, you have at least one copy of the table no matter what. And if only half of the operation succeeds, then you can clean it up by running DROP TABLE.

This way, if you lose connectivity after dropping the table, you may not be able to recover.